rollout/trafficrouting.go

```diff
@@ -98,6 +98,10 @@ func (c *rolloutContext) reconcileTrafficRouting() error {
 	desiredWeight := int32(0)
 	if c.rollout.Status.StableRS == c.rollout.Status.CurrentPodHash {
 		// when we are fully promoted. desired canary weight should be 0
+	} else if c.pauseContext.IsAborted() {
+		// when promote aborted. desired canary weight should be 0
+	} else if c.newRS.Status.AvailableReplicas == 0 {
+		// when newRS available replicas num is 0. never weight to canary
```

Comment:
This change is significant, and a surprising oversight if we are not handling this correctly. I will take some time to understand the implications of the change.